### PR TITLE
add azure_key_vault module and example

### PR DESCRIPTION
**Pull request by hesamwls**

### CODE REVIEW

File: modules/azurerm/key_vault/variables.tf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+variable location {
+  type        = "string"
+  description = "(Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created. For a list of all Azure locations, please consult this link: https://azure.microsoft.com/en-us/regions/"
+}
+
+variable key_perms {
+  type        = "list"
+  description = "(Optional) List of key operations client can perform. must be one or more from the following: backup, create, decrypt, delete, encrypt, get, import, list, purge, recover, restore, sign, unwrapKey, update, verify and wrapKey."
+  default     = ["backup", "create", "decrypt", "delete", "encrypt", "get", "import", "list", "recover", "restore", "sign", "unwrapKey", "verify", "wrapKey"]
```
Comment (ouzklcn) [inline modules/azurerm/key_vault/variables.tf]: Do you think using nearly all possible values on default is necessary? Maybe we should provide the least possible default values and let the user choose what he wants if he wants more or not provide any default value at all.

File: modules/azurerm/key_vault/variables.tf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+variable secret_perms {
+  type        = "list"
+  description = "(Optional) List of secret operations client can perform. Must be one or more form the following: backup, delete, get, list, purge, recover, restore and set. "
+  default     = ["backup", "delete", "get", "list", "recover", "restore", "set"]
```
Comment (ouzklcn) [inline modules/azurerm/key_vault/variables.tf]: Same question as above.

File: modules/azurerm/key_vault/main.tf [hunk 3]
```diff
@@ -0,0 +1,24 @@
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_key_vault" "key_vault" {
+  name     = "${var.name}-kv"
+  location = "${var.location}"
+  tags     = "${var.tags}"
+  resource_group_name = "${var.resource_group_name}"
```
Comment (ouzklcn) [inline modules/azurerm/key_vault/main.tf]: A `terraform fmt` command can fix indentation problems on tf files. It would be good if everything looks good.

Comment (hesamwls) [reply modules/azurerm/key_vault/main.tf]: cool, didn't know that.

File: modules/azurerm/key_vault/variables.tf (hunk 1 above)
Comment (hesamwls) [reply modules/azurerm/key_vault/variables.tf]: Yeah, I think all of them are non-harmful i.e. can't mess things up. only "purge" and "update" operations can shoot you in the head.

**Pull request merged**